week8_week9/prefix_tree/spec/node_spec.rb

```diff
@@ -0,0 +1,343 @@
+
+      it "other_child_node is part of nodes" do
+        expect(Node.nodes.include?(other_child_node)).to be_truthy
+      end
+    end
+  end
+
+  describe "#to_s" do
+    subject(:node) { described_class.new('t') }
+
+    context "to_s is puts values" do
+      subject(:first_child_node) { node.create_child('m') }
+      subject(:second_child_node) { node.create_child('d') }
+      subject(:not_child_node) { described_class.new('d') }
+
+      it "puts nodes children" do
+        expect(first_child_node.to_s).to eq('m')
+      end
+
+      it "puts nodes children" do
+        expect(second_child_node.to_s).to eq('d')
+      end
+
+      it "puts nodes children" do
+        expect(node.to_s).to eq('t')
+      end
+    end
+  end
+
+  describe "#include?" do
+    subject(:node) { described_class.new('t') }
+
+    context "check include? method" do
+      subject(:first_child_node) { node.create_child('m') }
+      
+      it "node children include first_child_node" do
+        expect(node.include?(first_child_node.to_s)).to be_truthy
+      end
+
+      subject(:second_child_node) { node.create_child('d') }
+      
+      it "node children include second_child_node" do
+        expect(node.include?(second_child_node.to_s)).to be_truthy
+      end
+
+      subject(:not_child_node) { described_class.new('d') }
+
+      it "node children not include not_child_node" do
+        expect(node.include?(not_child_node.to_s)).to be_falsey
+      end
+    end
+  end
+
+  describe "#find" do
+    subject(:node) { described_class.new('t') }
+
+    context "check find function" do
+      subject(:first_child_node) { node.create_child('m') }
+      subject(:second_child_node) { node.create_child('d') }
+      subject(:not_child_node) { described_class.new('d') }
+
+      it "return first_child_node" do
+        expect(node.find(first_child_node.to_s)).to equal(first_child_node)
+      end
+
+      it "return second_child_node" do
+        expect(node.find(second_child_node.to_s)).to equal(second_child_node)
+      end
+
+      it "return first_child_node" do
+        expect(node.find(not_child_node.to_s)).to be_falsey
+      end
+    end
+  end
+
+  describe "#delete_node" do
+    subject(:node) { described_class.new('t') }
+
+    context "check delete_node method" do
+      subject(:first_child_node) { node.create_child('m') }
+      subject(:second_child_node) { node.create_child('d') }
+      subject(:not_child_node) { described_class.new('d') }
+
+      it "delete_node from node" do
+        second_child_node.increment_size
+        node.delete_node
+        expect(node.children).to eq([second_child_node])
+      end
+
+      it "delete_node from node" do
+        node.delete_node
+        expect(node.children).to be_empty
+      end
+    end
+  end
+
+  describe "#become_end_point" do
+    subject(:node) { described_class.new('t') }
+
+    context "check become_end_point method" do
+      subject(:first_child_node) { node.create_child('m') }
```

Comment:
if you want to just add data use `before`
https://relishapp.com/rspec/rspec-core/v/2-0/docs/hooks/before-and-after-hooks